### ISSUE TITLE
Game locks up randomly

### ISSUE THREAD

**Issue by gamax92** [gamax92/wine-overwatch#2]
Overwatch will occasionally lock up, with these message printed to the terminal:

```
err:ntdll:RtlpWaitForCriticalSection section 0x1423b0970 "?" wait timed out in thread 004c, blocked by 0051, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x1423b0970 "?" wait timed out in thread 0009, blocked by 0051, retrying (60 sec)
```

Seems to be an issue with a recent version of Overwatch, as all wine versions (2.3 to 2.6) experience this freezing issue (though only 2.6 is giving me a message?)

EDIT: Select 'Windows XP' in winecfg to avoid the crash. https://bugs.winehq.org/show_bug.cgi?id=42741#c37

**Comment by gamax92** [comment]
I'm no longer able to reproduce this issue using the method I have in my notes, multiple games will play just fine without locking up, so I'll close this issue for now.

**Comment by sambow23** [comment]
with Wine version 2.11 and 2.10, I get random crashes or permanent freezes in-game.

**Comment by sambow23** [comment]
Terminal Output when the game crashed: fixme:system:SystemParametersInfoW Unknown action: 116
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:wtsapi:WTSUnRegisterSessionNotification Stub 0x60060
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:win:UnregisterDeviceNotification (handle=0xcafeaffe), STUB!
fixme:win:UnregisterDeviceNotification (handle=0xcafeaffe), STUB!
fixme:iphlpapi:CancelIPChangeNotify (overlapped 0x6b48d5c): stub
fixme:ntdll:EtwEventUnregister (deadbeef) stub.
fixme:font:RemoveFontMemResourceEx (0x877bdd19) stub
fixme:font:RemoveFontMemResourceEx (0x87454741) stub
fixme:font:RemoveFontMemResourceEx (0x81c04311) stub
fixme:msvcrt:__clean_type_info_names_internal (0x5a0e814) stub
fixme:win:UnregisterDeviceNotification (handle=0xcafecafe), STUB!
fixme:ntdll:EtwEventUnregister (deadbeef) stub.
fixme:ntdll:EtwEventUnregister (0) stub.
fixme:ntdll:EtwEventUnregister (deadbeef) stub.
fixme:ntdll:EtwEventUnregister (0) stub.
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 7
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 7
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 6
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 6
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 7
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 7
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 7
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 6
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 7
fixme:dbghelp:interpret_function_table_entry PUSH_MACHFRAME 6
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:keyboard:X11DRV_ActivateKeyboardLayout 0x4090409, 0000: semi-stub!
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2
fixme:msvcp:_Mtx_init_in_situ unknown flags ignored: 2

**Comment by zekesonxx** [comment]
@sambow23 What Windows version is Wine set to?

**Comment by gamax92** [comment]
Please do not randomly post things where they do not belong, you issue is not related to this issue report, create a new issue on github.

**Comment by Mad-Professor** [comment]
Game locks up about 5-10 seconds after returning to home screen.

Reproduce steps:
winecfg set to windows 10 with CSMT and DXVA2 enabled.

Start battlenet and launch Overwatch. 
Play single quickplay to the end, at MVP/Stat screen click leave game in the upper right corner, return to home screen, wait about 5-10 seconds, game freezes forcing killall or kill -9 or end process in ksysguard.

`err:ntdll:RtlpWaitForCriticalSection section 0x14261ad68 "?" wait timed out in thread 01fb, blocked by 001a, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x14261ad68 "?" wait timed out in thread 01fa, blocked by 001a, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x1001f8 "../../../wine-overwatch/dlls/xaudio2_9/../xaudio2_7/xaudio_dll.c: IXAudio2Impl.lock" wait timed out in thread 001a, blocked by 0034, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x14261ad68 "?" wait timed out in thread 01fb, blocked by 001a, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x14261ad68 "?" wait timed out in thread 01fa, blocked by 001a, retrying (60 sec)
`

I'm using 
i5 2500k
8GB
1050 Ti
Manjaro 4.9.39 using nvidia 375.66.
Compiled with latest commit, using wine 2.14 staging.

**Comment by Mad-Professor** [comment]
Ok I think this is related to audio, every time I play solo, and leave game or move back into queue, I do not get sound when I return to the home screen and the game freezes I get the output above. But If I am grouped with friends I always get a sound at the home screen when searching for a game. The game runs stable when grouped with friends as intended.

perhaps you should look closer at this. 
t` err:ntdll:RtlpWaitForCriticalSection section 0x1001f8 "../../../wine-overwatch/dlls/xaudio2_9/../xaudio2_7/xaudio_dll.c: IXAudio2Impl.lock" wait timed out in thread 001a, blocked by 0034, retrying (60 sec)`
